### DISCUSSION TITLE
Is it currently possible to display streaming output outside of jupyter notebook? If so, how?

### DISCUSSION THREAD

**Discussion by calebgasser** [guidance-ai/guidance#567] [Q&A]
There might be something for this, or it might not be possible, but I can't seem to figure out how to capture/display the streaming output outside of a jupyter notebook. Is there a way to capture this output and process it myself instead of just getting a bunch of `<IPython.core.display.HTML object>` objects back?

**Comment by slundberg** [comment]
We just merged support for programmatic streaming in #568. So as of `v0.1.11` you can now do:
```python
import guidance
from guidance import models, gen

lm = models.Transformers('gpt2')

@guidance
def f(lm):
    lm += "Tell me a story about Python." + gen(max_tokens=20)
    return lm

for part in lm.stream() + f():
    print(part)
```

Open to PRs that make printing work nicely by default outside of the notebook env as well :)
